tests/src/Simple/Exceptions/Exceptions.cs

```diff
@@ -29,21 +31,37 @@ public static int Main()
             }
             catch (OutOfMemoryException)
             {
+                Console.WriteLine("Unexpected exception caught");
                 return Fail;
             }
         }
         catch
```

Comment:
Fixed. Also added a smoke test for filters that uncovered a bug in the compiler.